crates/swc/src/plugin.rs

```diff
@@ -57,17 +80,22 @@ impl RustPlugins {
         if let Some(plugins) = &self.plugins {
             for p in plugins {
                 let config_json = serde_json::to_string(&p.1)
-                    .context("failed to serialize plugin config as json")?;
-                let config_json = Serialized::serialize(&config_json)?;
+                    .context("Failed to serialize plugin config as json")
+                    .and_then(|value| Serialized::serialize(&value))?;
+
+                let context_json = serde_json::to_string(&self.plugin_context)
```

Comment:
Is there a reason you selected json instead of `rkyv`?

Reply:
I wanted this to be json-based string, instead of strictly typed one. PR body tried to explain those (2 vs. 3). 

While I'm in favor of strict typed with rkyv, this'll make some tricky cases like

```
struct PluginStruct { 
 filename: string
}

... some times later

// This'll be breaking change!
struct PluginContext {
  filename:String;
  another: String
}
```

just adding a new property makes whole swc as breaking major bump. It is same as our AST cases. In this PR, instead makes context as string-based JSON and plugin author to pick up required property only which allows to add new properties without breaking changes. Still, this doesn't prevent removing / changing existing property with breaking changes though.

As wrote in comment (https://github.com/swc-project/swc/pull/3568#issuecomment-1039709771) interfacewise I prefer to have strict typed. If we want to go for those route (and I think we can), we need to define type of PluginContext reasonably suffecient while we do experiment / testing.

Thoughts?

Reply:
I think you are right, thanks!
I didn't think about breaking changes.